astrocut/footprint_cutout.py

```diff
@@ -228,27 +234,50 @@ def ra_dec_crossmatch(all_ffis: Table, coordinates: SkyCoord, cutout_size, arcse
         coordinates = SkyCoord(coordinates, unit='deg')
     ra, dec = coordinates.ra, coordinates.dec
 
-    # Parse cutout size
-    cutout_size = Cutout._parse_size_input(cutout_size)
-
-    # Create polygon for intersection
-    # Convert dimensions from pixels to arcseconds and divide by 2 to get offset from center
-    ra_offset = ((cutout_size[0] * arcsec_per_px) / 2) * u.arcsec
-    dec_offset = ((cutout_size[1] * arcsec_per_px) / 2) * u.arcsec
-
-    # Calculate RA and Dec boundaries
-    ra_bounds = [ra - ra_offset, ra + ra_offset]
-    dec_bounds = [dec - dec_offset, dec + dec_offset]
-
-    # Get RA and Dec for four corners of rectangle
-    ras = [ra_bounds[0].value, ra_bounds[1].value, ra_bounds[1].value, ra_bounds[0].value]
-    decs = [dec_bounds[0].value, dec_bounds[0].value, dec_bounds[1].value, dec_bounds[1].value]
-
-    # Create SphericalPolygon for comparison
-    cutout_fp = SphericalPolygon.from_radec(ras, decs, center=(ra, dec))
-
-    # Find indices of FFIs that intersect with the cutout
-    ffi_inds = np.vectorize(lambda ffi: ffi.intersects_poly(cutout_fp))(all_ffis['polygon'])
-    ffi_inds = FootprintCutout._ffi_intersect(all_ffis, cutout_fp)
+    px_size = np.zeros(2, dtype=object)
+    for axis, size in enumerate(Cutout._parse_size_input(cutout_size, allow_zero=True)):
+        if isinstance(size, u.Quantity):  # If Quantity, convert to pixels
+            if size.unit == u.pixel:
+                px_size[axis] = size.value
+            else:  # Angular size
+                # Convert angular size to pixels
+                px_size[axis] = (size.to_value(u.arcsec)) / arcsec_per_px
+        else:  # Assume pixels
+            px_size[axis] = size
+
+    if np.all(px_size == 0):
+        # A single point
+        ffi_inds = []
+        vector_coord = radec_to_vector(ra, dec)
+        for sector in np.unique(all_ffis['sequence_number']):
+            # Returns a 2-long array where the first element is indexes and the 2nd element is empty
+            sector_ffi_inds = np.where(all_ffis['sequence_number'] == sector)[0]
+
+            for ind in sector_ffi_inds:
+                if all_ffis[ind]["polygon"].contains_point(vector_coord):
+                    ffi_inds.append(ind)
+                    break  # the ra/dec will only be on one ccd per sector
+    else:
```

Comment:
as the size of this function has grown, i think it would be a good idea to split these two things out into separate functions:

- `_crossmatch_point`
- `_crossmatch_polygon`